### PR TITLE
Drop support for Python 2.7, 3.5, 3.6

### PR DESCRIPTION
**Pull request by dataflake**

### CODE REVIEW

**Review by gforcada — APPROVED** [review]

File: src/Products/SiteErrorLog/SiteErrorLog.py [hunk 1]
```diff
@@ -249,7 +235,7 @@ def _do_copy_to_zlog(self, now, strtype, entry_id, url, tb_text):
                             now - _rate_restrict_burst * _rate_restrict_period)
             next_when += _rate_restrict_period
             _rate_restrict_pool[strtype] = next_when
-            LOG.error('%s %s\n%s' % (entry_id, url, tb_text.rstrip()))
+            LOG.error(f'{entry_id} {url}\n{tb_text.rstrip()}')
```
Comment (mauritsvanrees) [inline src/Products/SiteErrorLog/SiteErrorLog.py:238]: For logging statements, I think it is still better to use the percentage syntax: it avoids needless formatting when logging for this level is switched off.
But if this is done with an automatic code formatter, like `pyupgrade` I wouldn't bother checking this.

I thought pyupgrade was smart about this though.
Ah, I see: in the original code it is already first parsed as string, and then passed to `LOG.error` without extra arguments. So the original code was not optimal.
Could be fixed, but I don't mind.

**Review by mauritsvanrees — APPROVED** [review]
LGTM.
One minor inline comment, but can be ignored.

File: src/Products/SiteErrorLog/SiteErrorLog.py (hunk 1 above)
Comment (dataflake) [reply src/Products/SiteErrorLog/SiteErrorLog.py:238]: IMHO it's better to be consistent, and in this case, who would switch off level `ERROR`? I would agree if this was something like `DEBUG` or `TRACE`, though.

**Comment by d-maurer** [comment]
Jens Vagelpohl wrote at 2023-2-1 02:14 -0800:
> ...
>IMHO it's better to be consistent, and in this case, who would switch off level `ERROR`? I would agree if this was something like `DEBUG` or `TRACE`, though.

I would consistently use logging calls with pattern and
pattern arguments -- and consistently avoid f-strings in this case.

**Comment by dataflake** [comment]
I would argue that most people would not know why that distinction is made and just be confused.

**Comment by d-maurer** [comment]
Jens Vagelpohl wrote at 2023-2-1 03:13 -0800:
>I would argue that most people would not know why that distinction is made and just be confused.

Many people learn from examples -- and a bit of confusion can
help in the process.

It is good to see no f-strings in logging calls. As pointed out
earlier, they add conputation costs which are wasted if
the logging configuration ignores the message.
Of course, you are right that this matters most for messages
at low levels (like "DEBUG") and are of less importance
for messages at level "ERROR" or higher.
But for consistency (and to provide good example code),
it is not bad to avoid f-strings in logging calls.

**Comment by dataflake** [comment]
I'll keep that in mind

**Pull request merged**